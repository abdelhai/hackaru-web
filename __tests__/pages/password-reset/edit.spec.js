import { Store } from 'vuex-mock-store';
import Factory from '@/__tests__/__setups__/factory';
import Edit from '@/pages/password-reset/edit';
import snakecaseKeys from 'snakecase-keys';

describe('Edit', () => {
  let factory;
  let wrapper;

  const $store = new Store();

  beforeEach(() => {
    $store.reset();
    factory = new Factory(Edit, {
      mocks: {
        $store,
        $route: {
          query: snakecaseKeys({
            userId: 1,
            token: 'passwordResetToken'
          })
        }
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when submit', () => {
    beforeEach(() => {
      $store.dispatch.mockReturnValue(true);
      wrapper = factory.shallow();
      wrapper.setData({
        password: 'password',
        passwordConfirmation: 'passwordConfirmation'
      });
      wrapper.find('form').trigger('submit.prevent');
    });

    it('dispatch auth/resetPassword', () => {
      expect($store.dispatch).toHaveBeenCalledWith('auth/resetPassword', {
        id: 1,
        password: 'password',
        passwordConfirmation: 'passwordConfirmation',
        token: 'passwordResetToken'
      });
    });

    it('redirect to /auth', () => {
      expect(factory.options.mocks.$router.push).toHaveBeenCalledWith(
        '/en/auth'
      );
    });
  });

  describe('when submit but failed', () => {
    beforeEach(() => {
      $store.dispatch.mockReturnValue(false);
      wrapper = factory.shallow();
      wrapper.find('form').trigger('submit.prevent');
    });

    it('does not redirect to /auth', () => {
      expect(factory.options.mocks.$router.push).not.toHaveBeenCalled();
    });
  });
});
