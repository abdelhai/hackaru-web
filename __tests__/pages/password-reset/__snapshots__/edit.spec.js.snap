// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit render correctly 1`] = `
<section>
  <heading-stub
    class="is-large"
  >
    
    title
  
  </heading-stub>
   
  <form>
    <text-field-stub
      aria-label="password"
      class="has-underline"
      placeholder="password"
      required=""
      type="password"
      value=""
    />
     
    <text-field-stub
      aria-label="passwordConfirmation"
      class="has-underline"
      placeholder="passwordConfirmation"
      required=""
      type="password"
      value=""
    />
     
    <btn-stub
      class="button is-rounded is-primary"
      type="submit"
    >
      
      reset
    
    </btn-stub>
  </form>
</section>
`;
